File: python/GreatestCommonDivisor.py
from random import *

def naive(a, b):
	if b == 0:
		return a

	currentGcd = 1
	
	for d in range(2, min(a, b) + 1):
		if a % d == 0 and b % d == 0:
			currentGcd = d

	return currentGcd

def euclidean(a, b):
	if b == 0:
		return a

	return euclidean(b, a % b)

a, b = map(int, input().split())
# print(naive(a, b) if a > b else naive(b, a))
print(euclidean(a, b) if a > b else euclidean(b, a))

'''
n = int(input())

while 1:
	a = randint(1, n)
	b = randint(1, n)
	print(a, b)
	
	r1 = naive(a, b) if a > b else naive(b, a)
	r2 = euclidean(a, b) if a > b else euclidean(b, a)
	
	if r1 == r2:
		print(" - ok")
	else:
		print(" - fail")
		break
'''

File: c++/GreatestCommonDivisor.cpp
#include "stdafx.h"

using namespace std;

int naive(int a, int b)
{
	if (b == 0)
		return a;

	int currentGcd = 1;
	for (int d = 2; d <= a && d <= b; d++) {
		if (a % d == 0 && b % d == 0) {
			currentGcd = d;
		}
	}

	return currentGcd;
}

int euclidean(int a, int b)
{
	if (b == 0)
		return a;

	return euclidean(b, a%b);
}

int start()
{
	cout << "Hello--------------------------------" << endl;
	int a = 0, b = 0;
	cin >> a;
	cin >> b;

	cout << naive(a, b) << endl;
	cout << euclidean(a, b) << endl;

	return 0;
}

